lib/transport/transport-socket.c

```diff
@@ -167,7 +161,14 @@ log_transport_dgram_socket_read_method(LogTransport *s, gpointer buf, gsize bufl
       rc = recvmsg(self->super.fd, &msg, 0);
     }
   while (rc == -1 && errno == EINTR);
-  return _extract_from_msghdr_method(self, rc, &msg, aux);
+  rc = _extract_from_msghdr_method(self, rc, &msg, aux);
+
+  if (rc == 0)
+    {
+      /* DGRAM sockets should never return EOF, they just need to be read again */
+      rc = -1;
+      errno = EAGAIN;
+    }
```

Comment:
[note]

This change removed the `return` statement of this function. It gets added back in a subsequent commit, though.